### PR TITLE
Add documentation - Getting Started - Harmonization

### PR DESCRIPTION
**Pull request by Aschen**
https://deploy-preview-1834--doc-kuzzle.netlify.app/guides/getting-started/1-run-kuzzle/

Also I added the following page: [Write an Application](https://deploy-preview-1834--doc-kuzzle.netlify.app/guides/getting-started/6-write-application/)

### CODE REVIEW

File: doc/2/guides/getting-started/2-storing-and-accessing-data/index.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ Kuzzle organizes the data storage in 4 levels:
 
 An index brings together several collections, which in turn contain several documents, each of which is composed of several fields.
 
-The collections have mappings that characterize the way Elasticsearch will index the data for the search.
+The **collections have mappings** that characterize the way Elasticsearch will index the data for the search.
```
Comment (scottinet) [inline doc/2/guides/getting-started/2-storing-and-accessing-data/index.md:21]:
```suggestion
The **collections have mappings** that characterize the way Elasticsearch will index the data for searches.
```

**Review by scottinet — APPROVED** [review]

File: doc/2/guides/getting-started/1-quick-start/index.md [hunk 2]
```diff
@@ -87,7 +88,7 @@ Now visit [http://localhost:7512](http://localhost:7512) with your browser. You
 We can also use the [Admin Console](/core/2/some-page-about-ac) which allows to manage your data, your users and your rights.
 
 ::: info
-The Admin Console is a Single Page Application written in Vue.js and using the Javascript SDK.  
+The Admin Console is a [Single Page Application](https://www.wikiwand.com/en/Single-page_application) written in Vue.js and using the [Javascript SDK](/sdk/js/7).
```
Comment (xbill82) [inline doc/2/guides/getting-started/1-quick-start/index.md:91]:
```suggestion
The Admin Console is a [Single Page Application](https://www.wikipedia.org/en/Single-page_application) written in Vue.js and using the [Javascript SDK](/sdk/js/7).  
```

File: doc/2/guides/getting-started/2-storing-and-accessing-data/index.md [hunk 3]
```diff
@@ -18,7 +18,7 @@ Kuzzle organizes the data storage in 4 levels:
 
 An index brings together several collections, which in turn contain several documents, each of which is composed of several fields.
 
-The collections have mappings that characterize the way Elasticsearch will index the data for the search.
+The **collections have mappings** that characterize the way Elasticsearch will index the data for searches.
```
Comment (xbill82) [inline doc/2/guides/getting-started/2-storing-and-accessing-data/index.md:21]:
```suggestion
The **collections have mappings** that define the way Elasticsearch will index the data for searches.
```

File: doc/2/guides/getting-started/1-quick-start/index.md (hunk 2 above)
Comment (Aschen) [reply doc/2/guides/getting-started/1-quick-start/index.md:91]: Ah yes! Ok but wikiwand is so good <3

**Review by xbill82 — APPROVED** [review]

File: doc/2/guides/getting-started/6-understand-application-development/index.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+::: warning
+Once the application has been started with the [Backend.start](/core/2/some-link) method, it is no longer possible to register new features.
+:::
+
+We will see in detail how to add controllers and pipes in the next chapters of this Getting Started.
+
+## Interact with the application
+
+Une fois l'application dans l'état démarrée, elle propose des méthodes pour intéragir avec votre application Kuzzle.
```
Comment (Njuelle) [inline doc/2/guides/getting-started/6-understand-application-development/index.md:69]: there's french here ;)

File: doc/2/guides/getting-started/6-understand-application-development/index.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+::: warning
+These methods interact directly with Kuzzle API and internal modules and are therefore not available until the application has been started with the [Backend.start](/core/2/some-link) method.
+:::
+
+### Embedded SDK
+
+In order to use the API actions, Kuzzle exposes the [Embedded SDK](/core/2/some-link) through the [Backend](/core/2/some-link) class.  
+
+You can acces it through the [Backend.sdk] property.
```
Comment (Njuelle) [inline doc/2/guides/getting-started/6-understand-application-development/index.md:97]: missing link?

**Review by Njuelle — CHANGES_REQUESTED** [review]

File: doc/2/guides/getting-started/6-understand-application-development/index.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+Calling the [Backend.start](/core/2/some-link) method will start your application and change it's state to `started`.
+:::
+
+There are two categories of functionalities:
+ - `registration`: available **before** starting the application
+ - `interaction`: available **after** starting the application
+
+## Register new features
+
+When my application is in the instanciated state, it exposes methods to **register new features** in Kuzzle core, like:
```
Comment (Njuelle) [inline doc/2/guides/getting-started/6-understand-application-development/index.md:39]:
```suggestion
When the application is in the instanciated state, it exposes methods to **register new features** in Kuzzle core, like:
```

**Review by jenow — APPROVED** [review]

File: doc/2/guides/getting-started/9-deploy-your-application/index.md [hunk 7]
```diff
@@ -1,16 +1,16 @@
...
 ---
 
-# Deploy
+# Deploy your Application
 
-Kuzzle is just a Node.js application that also needs Elasticsearch and Redis to run.  
+Kuzzle is just a Node.js application that also needs [Elasticsearch](https://www.elastic.co/what-is/elasticsearch) and [Redis](https://redis.io/topics/introduction) to run.  
 
-The only specifity is that Kuzzle needs to compile C and C++ dependencies so the `npm install` will also be looking for `python`, `make` and `g++` packages.   
+The only specifity is that **Kuzzle needs to compile C and C++ dependencies** so the `npm install` will also be looking for `python`, `make` and `g++` packages.
```
Comment (scottinet) [inline doc/2/guides/getting-started/9-deploy-your-application/index.md:13]:
```suggestion
The only specifity is that **Kuzzle needs to compile C and C++ dependencies** so the `npm install` will also be looking for the `python`, `make` and `g++` packages.   
```

**Review by scottinet — CHANGES_REQUESTED** [review]

File: doc/2/guides/getting-started/6-understand-application-development/index.md [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+You can access the following controllers from the API:
+- [auth](/sdk/js/7/controllers/auth)
+- [bulk](/sdk/js/7/controllers/bulk)
+- [collection](/sdk/js/7/controllers/collection)
+- [document](/sdk/js/7/controllers/document)
+- [index](/sdk/js/7/controllers/index)
+- [memoryStorage (ms)](/sdk/js/7/controllers/ms)
+- [security](/sdk/js/7/controllers/security)
+- [server](/sdk/js/7/controllers/server)
+- [realtime](/sdk/js/7/controllers/realtime)
```
[the remark this answers is not in the document]
Comment (Aschen) [reply doc/2/guides/getting-started/6-understand-application-development/index.md:112]: Actually every controllers can be accessed now :sweat_smile:

**Review by alexandrebouthinon — APPROVED** [review]
I know this is not in the current diff but we should format this WebSocket response https://github.com/kuzzleio/kuzzle/blame/56b77ceffa54fe525390c1cf57b9ad68afb8b332/doc/2/guides/getting-started/7-create-new-controllers/index.md#L90

File: doc/2/guides/getting-started/6-write-application/index.md [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+
+## Interact with the application
+
+Once the application has started, methods to interact with your Kuzzle application become available.
+
+In particular, those methods make it possible to:
+ - use the [Embedded SDK](/core/2/some-link) to use the API
+ - trigger events
+ - log messages
+ - and more..!
```
Comment (Leodau) [inline doc/2/guides/getting-started/6-write-application/index.md:75]:
```suggestion
 - Use the [Embedded SDK](/core/2/some-link) to use the API
 - Trigger Events
 - Log Messages
 - And much more!
```

**Review by Leodau — CHANGES_REQUESTED** [review]

File: doc/2/guides/getting-started/6-write-application/index.md [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+
+**Example:** Use the Embedded SDK to create an index after startup with the [index.create](/sdk/js/7/controllers/index/create) method
+```js
+app.start()
+  .then(async () => {
+    // interact with Kuzzle API to create a new index if it does not already exist
```
Comment (Leodau) [inline doc/2/guides/getting-started/6-write-application/index.md:81]:
```suggestion
    // Interact with Kuzzle API to create a new index if it does not already exist
```

File: doc/2/guides/getting-started/6-write-application/index.md [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+
+You can access it through the [Backend.sdk](/core/2/some-link) property.  
+
+::: info
+The Embedded SDK is a modified version of the [Javascript SDK](/sdk/js/7) which is directly connected to the API and does not send requests through the network.  
+:::
+
+**Example:** _Create a new document by using the [document.create](/sdk/js/7/controllers/document/create) method_
+```js
+// after application startup
```
Comment (Leodau) [inline doc/2/guides/getting-started/6-write-application/index.md:105]:
```suggestion
// After application startup
```

File: doc/2/guides/getting-started/6-write-application/index.md [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+
+// creates a document
```
Comment (Leodau) [inline doc/2/guides/getting-started/6-write-application/index.md:107]:
```suggestion
// Creates a document
```

File: doc/2/guides/getting-started/6-write-application/index.md [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+})
+```
+
+::: info 
+The low level [query](/sdk/js/7/core-classes/kuzzle/query) method can also be used to send custom requests to the Kuzzle API.  
+:::
+
+**Example:** _Execute a custom controller action with the [query](/sdk/js/7/core-classes/kuzzle/query) method_
+```js
+// after application startup
```
Comment (Leodau) [inline doc/2/guides/getting-started/6-write-application/index.md:120]:
```suggestion
// After application startup
```

File: doc/2/guides/getting-started/6-write-application/index.md [hunk 14]
```diff
@@ -0,0 +1,162 @@
...
+
+// execute a custom controller action
```
Comment (Leodau) [inline doc/2/guides/getting-started/6-write-application/index.md:122]:
```suggestion
// Execute a custom controller action
```

File: doc/2/guides/getting-started/6-write-application/index.md [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+  name: 'Aschen'
+})
+```
+
+## Complete example
+
+```js
+import { Backend } from 'kuzzle'
+
+// instantiate an application
```
Comment (Leodau) [inline doc/2/guides/getting-started/6-write-application/index.md:135]:
```suggestion
// Instantiate an application
```

File: doc/2/guides/getting-started/6-write-application/index.md [hunk 16]
```diff
@@ -0,0 +1,162 @@
...
+const app = new Backend('playground')
+
+// now we can register features
```
Comment (Leodau) [inline doc/2/guides/getting-started/6-write-application/index.md:138]:
```suggestion
// Now we can register features
```

**Comment by Aschen** [comment]
@scottinet Should we start list items by a capital letter?

```
Hello:
 - We are
 - In a List
```

Instead of
```
Hello:
 - we are
 - in a list
```

File: doc/2/guides/getting-started/6-write-application/index.md [hunk 17]
```diff
@@ -0,0 +1,162 @@
...
+
+// register a new controller
```
Comment (Leodau) [inline doc/2/guides/getting-started/6-write-application/index.md:140]:
```suggestion
// Register a new controller
```

File: doc/2/guides/getting-started/6-write-application/index.md [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+  actions: {
+    sayHello: {
+      handler: async request => {
+        return `Hello, ${request.input.args.name}`
+      }
+    }
+  }
+})
+
+// start the application
```
Comment (Leodau) [inline doc/2/guides/getting-started/6-write-application/index.md:151]:
```suggestion
// Start the application
```

File: doc/2/guides/getting-started/6-write-application/index.md [hunk 19]
```diff
@@ -0,0 +1,162 @@
...
+app.start()
+  .then(async () => {
+    // now we can interact with Kuzzle API
+
+    // interact with Kuzzle API to creates a new index if it does not exists
```
Comment (Leodau) [inline doc/2/guides/getting-started/6-write-application/index.md:156]:
```suggestion
    // Now we can interact with Kuzzle API to create a new index if it does not exists
```

**Review by Leodau — CHANGES_REQUESTED** [review]

File: doc/2/guides/getting-started/1-run-kuzzle/index.md [hunk 20]
```diff
@@ -20,6 +20,7 @@ In this guide we will use Docker and Docker Compose to run those services.
  - [Docker](https://docs.docker.com/engine/install/)
  - [Docker Compose](https://docs.docker.com/compose/install/)
  - [Kourou](https://github.com/kuzzleio/kourou)
+ - Compile toolchain: g++, make and python
```
Comment (scottinet) [inline doc/2/guides/getting-started/1-run-kuzzle/index.md:23]:
```suggestion
 - Compile toolchain: a C++ compiler (g++, clang, ...), make and python
```

File: doc/2/guides/getting-started/2-store-and-access-data/index.md [hunk 21]
```diff
@@ -1,27 +1,29 @@
...
 code: false
 type: page
-title: Storing and accessing Data
+title: Store and Access Data
 description: Create and retrieve documents
 order: 200
 ---
 
-# Storing and accessing Data
+# Store and access Data
```
Comment (scottinet) [inline doc/2/guides/getting-started/2-store-and-access-data/index.md:9]:
```suggestion
# Store and Access Data
```

File: doc/2/guides/getting-started/2-store-and-access-data/index.md [hunk 22]
```diff
@@ -34,6 +36,12 @@ If you're more familiar with the way relational databases store data, here is an
 | line                         | document                  |
 | column                       | field                     |
 
+::: info
+Kuzzle uses Elasticsearch as a NoSQL document storage.  
+Elasticsearch is primarly designed to be a search engine, so there are limitations when using it as a database.
```
Comment (scottinet) [inline doc/2/guides/getting-started/2-store-and-access-data/index.md:41]:
```suggestion
Elasticsearch is primarily designed to be a search engine, so there are limitations when using it as a database.  
```

File: doc/2/guides/getting-started/9-deploy-your-application/index.md [hunk 23]
```diff
@@ -1,16 +1,16 @@
...
 ---
 
-# Deploy
+# Deploy your Application
 
-Kuzzle is just a Node.js application that also needs Elasticsearch and Redis to run.  
+Kuzzle is just a Node.js application that also needs [Elasticsearch](https://www.elastic.co/what-is/elasticsearch) and [Redis](https://redis.io/topics/introduction) to run.  
 
-The only specifity is that Kuzzle needs to compile C and C++ dependencies so the `npm install` will also be looking for `python`, `make` and `g++` packages.   
+The only specifity is that **Kuzzle needs to compile C and C++ dependencies** so a successful `npm install` execution needs the `python`, `make` and `g++` packages.
```
Comment (scottinet) [inline doc/2/guides/getting-started/9-deploy-your-application/index.md:13]:
```suggestion
The only specifity is that **Kuzzle needs to compile C and C++ dependencies** so a successful `npm install` execution needs the `python` and `make` packages, as well as C and C++ compilers (in the following examples we'll use `g++`).   
```

**Review by scottinet — CHANGES_REQUESTED** [review]
a few minor changes, otherwise it looks good to me

**Review by scottinet — APPROVED** [review]

**Review by Leodau — APPROVED** [review]

**Pull request merged**